charms/candid/src/charm.py

```diff
@@ -234,47 +258,57 @@ def _on_master_changed(self, event: pgsql.MasterChangedEvent) -> None:
             "Updating database configuration...", WaitingStatus
         )
 
-        uri = event.master.uri
-        if not uri:
-            uri = ""
-        logging.info("database uri {}".format(uri))
+        if not event.master or not event.master.uri:
+            logging.debug("removing database connection string")
+            del self._state.db_uri
```

Comment:
Maybe in this case calling the stop hook and returning might be good?

Reply:
this hook calls _update_config_and_restart which will now stop the service should any config value be missing